Remove mm_config and mm_license global state from webapp (PR #6)

lieut-data opened the pull request:
#### Summary
This is a subset of the changes in the `MM-8589-remove_mm_global_license` branch, broken up into multiple PRs to facilitate easier review. See any linked PRs for other changes if you're interested. 

These changes, as a whole, move towards removing our dependence on the global `mm_config` and `mm_license` objects. These changes will ultimately facilitate resolving https://mattermost.atlassian.net/browse/MM-8604, allowing us to stop hard refreshing the application whenever someone makes a configuration change.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-8589

#### Checklist
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed
- [x] Added or updated unit tests (required for all new features)

### Related PRs
https://github.com/mattermost/mattermost-webapp/pull/816
https://github.com/mattermost/mattermost-webapp/pull/819
https://github.com/mattermost/mattermost-webapp/pull/820
https://github.com/mattermost/mattermost-webapp/pull/821
https://github.com/mattermost/mattermost-webapp/pull/822

jwilander commented on components/channel_notifications_modal/index.js:
```diff
@@ -0,0 +1,18 @@
+    const sendPushNotifications = config.SendPushNotifications === 'true';
+
+    return {
+        ...ownProps,
```
We actually don't need to include `...ownProps` in this return statement, the connect function handles it. I know we do it everywhere but it was a mistake that got copied/pasted everywhere before we caught it. @stephenkiers has a a PR #798 to remove them all

I'm 0/5 if you want to go through the effort of removing them here though, I'm trying to avoid being too nitpicky because of the volume of PRs coming through

lieut-data replied:
Ah, fascinating! I'll either rip them out as part of the PRs, or go back and fix things up later depending on the volume of feedback I receive.

jespino commented on components/channel_notifications_modal/channel_notifications_modal.jsx:
```diff
@@ -247,5 +247,6 @@ ChannelNotificationsModal.propTypes = {
     channelMember: PropTypes.object.isRequired,
-    currentUser: PropTypes.object.isRequired
+    currentUser: PropTypes.object.isRequired,
+    sendPushNotifications: PropTypes.bool.isRequired
```
We can move the propTypes to the `static propType` class attribute. Not a stopper.

jespino commented on tests/components/navbar/navbar.test.jsx:
```diff
@@ -55,7 +57,10 @@ describe('components/navbar/Navbar', () => {
     test('should match snapshot, if not licensed', () => {
         global.window.mm_license.IsLicensed = 'false';
```
I think this line (and similar ones in the following tests) are no longer needed.

lieut-data replied:
Unfortunately, there's some remaining cleanup for higher level components that makes these global assignment still necessary -- for now. Rest assured, it's all going away soon :)

jespino commented:
@lieut-data feel free to ask me for another review if you apply new changes and you think that changes need review. 🙂

lieut-data commented:
Thanks, @jespino! I've rebased and removed the unnecessary `ownProps` and converged on the `get(Config|License)` selectors. I think we're good to go, so I'll merge as soon as the build completes.
